### PR TITLE
Migrate TV.UIControls to MAUI

### PR DESCRIPTION
**Pull request by shyunMin**
### Description
This PR is to prepare migrating TV.UIControls to .NET MAUI.

### Issues
|Test|status|
|-|-| 
|Shadow Frame Test| done |
|Page Transition Test|  |
|MediaPlayer| |
|RemoteControl Test| |
|DrawerLayout Test| |
|Entry and Editor Test| done |
|CustomFocus Test| done |
|ContentButton Test| done |
|ContentPopup Test| |
|Grid Test| |

### CODE REVIEW

File: sample/Sample/GridView/GridViewModel.cs [hunk 1]
```diff
@@ -104,67 +105,71 @@ class PosterModel
...
+            //    "42 The Social Network.jpg",
+            //    "43 The Thing.jpg",
+            //    "44 Fear and Loathing in Las Vegas.jpg",
+            //    "45 Love In The Afternoon.jpeg",
+            //    "46 Manhattan.jpeg",
+            //    "47 Lord of War.jpg",
+            //    "48 Mean Streets.jpg",
+            //    "49 The Graduate.jpg",
+            //    "50 little miss sunshine.jpg"
+            //};
```
Comment (rookiejava) [inline sample/Sample/GridView/GridViewModel.cs]: Please remove unnecessary lines.

File: sample/Sample/GridView/GridViewModel.cs [hunk 2]
```diff
@@ -190,31 +200,36 @@ public static List<PosterModel> MakeAlbumArtsModel()
...
+                //"Back in Black, ACDC.jpg",
+                //"Elvis Presley, Elvis Presley.jpg",
+                //"Feel So Close, Calvin Harris.jpg",
+                //"Feel, Third Party vs Cicada.jpg",
+                //"I'm in Love (I Wanna Do It), Alex Gaudino.jpg",
+                //"If I Lose Myself, OneRepublic vs Alesso.jpg",
+                //"Nevermind, Nirvana.jpg",
+                //"Nothing's Real, Shura.jpg",
+                //"Runaway (U&I), Galantis.jpg",
+                //"The Freewheelin' Bob Dylan, Bob Dylan.jpg",
```
Comment (rookiejava) [inline sample/Sample/GridView/GridViewModel.cs]: ditto

File: sample/Sample/RecycleItemsView/RecycleItemsViewModel.cs [hunk 3]
```diff
@@ -258,15 +230,18 @@ public static List<PosterModel> MakeAlbumArtsModel()
...
             foreach (var i in albums)
             {
-                var texts = i.Split('.');
-                texts[texts.Length - 1] = "";
-                var text = string.Join(" ", texts).Trim();
-                texts = text.Split(',');
+                //var texts = i.Split('.');
+                //texts[texts.Length - 1] = "";
+                //var text = string.Join(" ", texts).Trim();
+                //texts = text.Split(',');
```
Comment (rookiejava) [inline sample/Sample/RecycleItemsView/RecycleItemsViewModel.cs]: ditto

File: src/Tizen.TV.UIControls.Forms/Focus.cs [hunk 4]
```diff
@@ -182,27 +182,27 @@ static void KeyHandler(RemoteControlKeyEventArgs args)
         }
     }
 
-    class PlatformFocusUpEffect : RoutingEffect
+    public class FocusUpEffect : RoutingEffect
```
Comment (rookiejava) [inline src/Tizen.TV.UIControls.Forms/Focus.cs]: Is there any reason to change to public?

File: src/Tizen.TV.UIControls.Forms/Focus.cs [hunk 5]
```diff
@@ -182,27 +182,27 @@ static void KeyHandler(RemoteControlKeyEventArgs args)
...
     {
-        public PlatformFocusUpEffect() : base("TizenTVUIControl.PlatformFocusUpEffect")
+        public FocusUpEffect() : base("TizenTVUIControl.PlatformFocusUpEffect")
```
Comment (rookiejava) [inline src/Tizen.TV.UIControls.Forms/Focus.cs]: Doesn't we need to change `effectID` as well?

**Review by rookiejava — COMMENTED** [review]
For better readability, please clean unnecessary codes and comments.

File: sample/Sample/GridView/GridViewModel.cs (hunk 1 above)
Comment (shyunMin) [reply sample/Sample/GridView/GridViewModel.cs]: I've removed unnecessary lines.

File: src/Tizen.TV.UIControls.Forms/Focus.cs (hunk 4 above)
Comment (shyunMin) [reply src/Tizen.TV.UIControls.Forms/Focus.cs]: Unlike `ExportEffect` attribute, accessing this class is also required when registering the effect.

File: src/Tizen.TV.UIControls.Forms/Focus.cs (hunk 5 above)
Comment (shyunMin) [reply src/Tizen.TV.UIControls.Forms/Focus.cs]: I've updated effectId

**Review by rookiejava — COMMENTED** [review]
This was big PR that had to be modified overall for the project to migrate to.NET MAUI. Thank you for hard work!.

The changes are big, so it may take more time to review. I'll look at it as soon as possible. 😸 
/cc @sung-su @myroot @JoonghyunCho Please review it.

And, I modified the ci build to be applied to the maui-migration branch and submitted in PR #94. I think it would be better to apply #94 before this PR for build test. What do you think?

File: sample/Sample/GridView/GridViewModel.cs (hunk 2 above)
Comment (rookiejava) [reply sample/Sample/GridView/GridViewModel.cs]: I confirmed that it has been modified. 👍

File: sample/Sample/RecycleItemsView/RecycleItemsViewModel.cs (hunk 3 above)
Comment (rookiejava) [reply sample/Sample/RecycleItemsView/RecycleItemsViewModel.cs]: I confirmed that it has been modified. 👍

File: src/Tizen.TV.UIControls.Forms/Focus.cs (hunk 4 above)
Comment (rookiejava) [reply src/Tizen.TV.UIControls.Forms/Focus.cs]: got it. Thank you for your explanation.

What do you think about separating this class in terms of "one public class per file". Of course, I don't think this rule is compulsory, but especially if it's a public class, we used to follow this rule for the ease of managing the public API.

**Comment by JoonghyunCho** [comment]
@rookiejava Absolutely a great idea! Thanks for updating the CI for maui migration :)

**Comment by rookiejava** [comment]
@shyunMin Could you rebase onto latest maui-migrate for applying ci build test.

File: src/Tizen.Theme.Common/GridView.cs [hunk 6]
```diff
@@ -149,9 +153,9 @@ public object SelectedItem
             set { SetValue(SelectedItemProperty, value); }
         }
 
-        /// <summary>
```
Comment (JoonghyunCho) [inline src/Tizen.Theme.Common/GridView.cs]: additional `//` is added here

File: src/Tizen.Theme.Common/Renderer/CommonUI.cs [hunk 7]
```diff
@@ -27,30 +27,30 @@ namespace Tizen.Theme.Common
...
         /// </summary>
-        public CoreApplication Context { get; set; }
+        public MauiApplication Context { get; set; }
 
         /// <summary>
         /// Gets or sets the main window provider
         /// </summary>
         public Func<Window> MainWindowProvider { get; set; }
 
-        /// <summary>
```
Comment (JoonghyunCho) [inline src/Tizen.Theme.Common/Renderer/CommonUI.cs]: additional `//` is added here too

File: src/Tizen.TV.UIControls.Forms/Focus.cs (hunk 4 above)
Comment (shyunMin) [reply src/Tizen.TV.UIControls.Forms/Focus.cs]: Thank you for your suggestion.
I agree that it is better to separate the file for each public classes and I've fixed it.

File: src/Tizen.Theme.Common/GridView.cs (hunk 6 above)
Comment (shyunMin) [reply src/Tizen.Theme.Common/GridView.cs]: Fixed.

File: src/Tizen.Theme.Common/Renderer/CommonUI.cs (hunk 7 above)
Comment (shyunMin) [reply src/Tizen.Theme.Common/Renderer/CommonUI.cs]: Fixed.

**Comment by shyunMin** [comment]
@rookiejava I've upated and rebased this patch with applying review feedback :)

**Comment by rookiejava** [comment]
> @rookiejava I've upated and rebased this patch with applying review feedback :)

Yay! CI build integraion has been completed. 🎉 

However, as you can see from the link below, there is a lot of warning.
https://github.com/Samsung/Tizen.TV.UIControls/runs/6826926568?check_suite_focus=true#step:6:215

As mentioned above, this PR is too big, so it would be better to remove the warnings with a separate PR. What do you think?

```sh
Warning: /home/runner/work/Tizen.TV.UIControls/Tizen.TV.UIControls/src/Tizen.Theme.Common/OverlayMediaView.cs(67,92): warning CS8625: Cannot convert null literal to non-nullable reference type. [/home/runner/work/Tizen.TV.UIControls/Tizen.TV.UIControls/src/Tizen.Theme.Common/Tizen.Theme.Common.csproj]
Warning: /home/runner/work/Tizen.TV.UIControls/Tizen.TV.UIControls/src/Tizen.Theme.Common/Renderer/AnimatedNavigationPageRenderer.cs(30,51): warning CS0618: 'VisualElementRenderer<AnimatedNavigationPage>' is obsolete: 'Use Microsoft.Maui.Controls.Handlers.Compatibility.VisualElementRenderer instead' [/home/runner/work/Tizen.TV.UIControls/Tizen.TV.UIControls/src/Tizen.Theme.Common/Tizen.Theme.Common.csproj]
Warning: /home/runner/work/Tizen.TV.UIControls/Tizen.TV.UIControls/src/Tizen.Theme.Common/Renderer/GridViewRenderer.cs(36,37): warning CS0618: 'ViewRenderer<GridView, GenGrid>' is obsolete: 'Use Microsoft.Maui.Controls.Handlers.Compatibility.ViewRenderer instead' [/home/runner/work/Tizen.TV.UIControls/Tizen.TV.UIControls/src/Tizen.Theme.Common/Tizen.Theme.Common.csproj]
Warning: /home/runner/work/Tizen.TV.UIControls/Tizen.TV.UIControls/src/Tizen.Theme.Common/Renderer/MediaViewRenderer.cs(24,38): warning CS06[18](https://github.com/Samsung/Tizen.TV.UIControls/runs/6826926568?check_suite_focus=true#step:6:19): 'ViewRenderer<MediaView, LayoutCanvas>' is obsolete: 'Use Microsoft.Maui.Controls.Handlers.Compatibility.ViewRenderer instead' [/home/runner/work/Tizen.TV.UIControls/Tizen.TV.UIControls/src/Tizen.Theme.Common/Tizen.Theme.Common.csproj]
Warning: /home/runner/work/Tizen.TV.UIControls/Tizen.TV.UIControls/src/Tizen.Theme.Common/Renderer/OverlayPageRenderer.cs(21,40): warning CS0618: 'PageRenderer' is obsolete: 'This renderer is obsolete please try to use the corresponding handler/mapper' [/home/runner/work/Tizen.TV.UIControls/Tizen.TV.UIControls/src/Tizen.Theme.Common/Tizen.Theme.Common.csproj]
Warning: /home/runner/work/Tizen.TV.UIControls/Tizen.TV.UIControls/src/Tizen.Theme.Common/Renderer/OverlayViewRenderer.cs(27,40): warning CS0618: 'ViewRenderer<OverlayMediaView, LayoutCanvas>' is obsolete: 'Use Microsoft.Maui.Controls.Handlers.Compatibility.ViewRenderer instead'
```

**Comment by shyunMin** [comment]
Yes I agreed this PR is a giant step as it contains a lots of minor updates to migrate to MAUI. I will remove warnings and update issues on the following PRs. Thanks!😊

**Review by JoonghyunCho — APPROVED** [review]
Thanks for the big step forward on the MAUI Tizen!

**Review by rookiejava — APPROVED** [review]
Thank you for your effort!

**Pull request merged**